mautrix_telegram/portal.py

```diff
@@ -2318,6 +2342,18 @@ async def _handle_telegram_document(
                 "image/": MessageType.IMAGE,
             }.get(info.mimetype[:6], MessageType.FILE),
         )
+        if attrs.is_audio:
+            content["org.matrix.msc1767.file"] = {
+                "url": content.url,
+                "name": content.body,
+                **(content.file.serialize() if content.file else {}),
+                **(content.info.serialize() if content.info else {}),
+            }
+            content["org.matrix.msc1767.audio"] = {"duration": attrs.duration}
+            if attrs.waveform:
+                content["waveform"] = [int(x) for x in attrs.waveform]
```

Comment:
Since Telegram uses 5 bits and Matrix uses 0-1024 (which is 10 bits + 1), we probably also need to shift it left by 5 (or multiply by 32):

```suggestion
                content["org.matrix.msc1767.audio"]["waveform"] = [x << 5 for x in attrs.waveform]
```